processing/src/main/java/org/apache/druid/query/aggregation/first/FloatFirstAggregatorFactory.java

```diff
@@ -281,7 +236,7 @@ public byte[] getCacheKey()
   public ColumnType getIntermediateType()
   {
     // if we don't pretend to be a primitive, group by v1 gets sad and doesn't work because no complex type serde
```

Comment:
This can be removed from all the numeric factories

Reply:
Wont this throw an exception as `AggregateFactory::getIntermediateType` default implementation calls `getType` which throws `UnsupportedOperationException`  and looks like the doc says `getType` will be deprecated in support of `getIntermediateType`

Reply:
This was aimed at the comment, apologies for being unclear the first time. Currently intermediate types just return the primitive, which I got to know was a hack for making it work sometimes (after discussion with Gian). 
Now we have the correct complex types therefor we can clean out this comment.

```suggestion

```